A modification of a Gaussian process is a Gaussian process

EtienneC30 opened the pull request:
I also generalize the instance saying that a Gaussian measure is a probability measure and prove that a Gaussian process is almost everywhere measurable. I need to add this hypothesis to the theorem. Indeed if `Y` is a modification of `X` and `X` is a Gaussian process, then `Y` might fail to be almost everywhere measurable (only if `T` is not countable), in which case it would not be a Gaussian process (see [Zulip](https://leanprover.zulipchat.com/#narrow/channel/509433-Brownian-motion/topic/Tasks.20and.20claims/near/525387577)).
I also generalize `finite_distributions_eq` to an arbitrary fintype. Should I do the same for other lemmas in the same file, or should keep the current version and duplicate this particular lemma?

RemyDegenne commented:
- about generalizing beyond normed spaces: fine, it can't hurt. Is there any use of the notion "Gaussian" outside of Banach spaces though?
- AEMeasurable hypotheses: :+1:  This project made me realize that for `Measurable`, the pointwise and global properties are equivalent, but not for `AEMeasurable` in general.
- about generalizing to an arbitrary fintype: feel free to do it everywhere!

EtienneC30 commented:
Ok I changed the definition of a Gaussian process to
```
def IsGaussianProcess (X : T → Ω → E) (P : Measure Ω := by volume_tac) : Prop :=
  ∀ I : Finset T, IsGaussian (P.map (fun ω ↦ I.restrict (X · ω)))
```
and I rewritten `StochasticProcesses` to use finsets instead of maps from fintypes. I think this was the right call because the proofs got simpler. Also I removed the use of `measurability` which made the file slow.

RemyDegenne reviewed (approved):
Thanks!
(I'll merge in around 30min, when CI for the previous merged PR is done. The doc step of CI does not manage to store a cache for some reason, so it's super slow)
